include/linux/sched.h

```diff
@@ -1315,6 +1316,16 @@ struct task_struct {
 	struct callback_head		mce_kill_me;
 #endif
 
+ //needed in double fetch detection buffer to store address ranges 
+#ifdef CONFIG_DF_DETECTION
```

Comment:
Please declare a struct in linux/df-detection.h that will hold all these fields, so that task_struct contains just an instance of that struct instead of assorted data.